File: Data Structures & Algorithms/daily-temperatures/submission-3.js
class Solution {
    /**
     * @param {number[]} temp
     * @return {number[]}
     */
    dailyTemperatures(temp) {
        const len = temp.length;
        const ans = new Array(len).fill(0);
        let max = temp[len-1];
        
        for (let i=len-2; i>=0; i--) {
            if (temp[i] < max) {
                let findIdx = i;
                while (findIdx<=len-1){
                    if (temp[findIdx] > temp[i]) {
                        ans[i] = findIdx-i;
                        break
                    }
                    findIdx++;
                }
            } else {
                max = temp[i];
            }
        }
        return ans
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = prices[0]
        maxProfit = 0

        for price in prices[1:]:
            if l >= price:
                l = price
            else:
                maxProfit = max(maxProfit, price-l)

        return maxProfit